endpoints/api/policy.py

```diff
@@ -0,0 +1,323 @@
+            raise Unauthorized()
+
+        app_data = request.get_json()
+        method = app_data.get("method", None)
+        value = app_data.get("value", None)
+
+        if method is None or value is None:
+            request_error(message="Missing the following parameters: method, value")
+
+        policy_config = {
+            "method": method,
+            "value": value,
+        }
+
+        try:
+            policy = model.autoprune.create_namespace_autoprune_policy(
+                orgname, policy_config, create_task=True
+            )
+        except model.InvalidNamespaceException:
+            raise NotFound()
+        except model.InvalidNamespaceAutoPrunePolicy as ex:
+            request_error(ex)
+        except model.NamespaceAutoPrunePolicyAlreadyExists as ex:
+            request_error(ex)
+
+        return {"uuid": policy.uuid}, 201
+
+
+@resource("/v1/organization/<orgname>/autoprunepolicy/<policy_uuid>")
+@path_param("orgname", "The name of the organization")
+@path_param("policy_uuid", "The unique ID of the policy")
+@show_if(features.AUTO_PRUNE)
+class OrgAutoPrunePolicy(ApiResource):
+    """
+    Resource for fetching, updating, and deleting specific organization auto-prune policies
+    """
+
+    schemas = {
+        "AutoPrunePolicyConfig": {
+            "type": "object",
+            "description": "The policy configuration that is to be applied to the organization",
+            "required": ["method", "value"],
+            "properties": {
+                "method": {
+                    "type": "string",
+                    "description": "The method to use for pruning tags (number_of_tags, creation_date)",
+                },
+                "value": {
+                    "type": ["integer", "string"],
+                    "description": "The value to use for the pruning method (number of tags e.g. 10, time delta e.g. 7d (7 days))",
+                },
+            },
+        },
+    }
+
+    def get(self, orgname, policy_uuid):
+        permission = AdministerOrganizationPermission(orgname)
+        if not permission.can() and not allow_if_superuser():
+            raise Unauthorized()
+
+        policy = model.autoprune.get_namespace_autoprune_policy(orgname, policy_uuid)
+        if policy is None:
+            raise NotFound()
+
+        return policy.get_view()
+
+    @validate_json_request("AutoPrunePolicyConfig")
+    def put(self, orgname, policy_uuid):
+        permission = AdministerOrganizationPermission(orgname)
+        if not permission.can() and not allow_if_superuser():
+            raise Unauthorized()
+
+        app_data = request.get_json()
+        method = app_data.get("method", None)
+        value = app_data.get("value", None)
+
+        if method is None or value is None:
+            request_error(message="Missing the following parameters: method, value")
+
+        policy_config = {
+            "method": method,
+            "value": value,
```

Comment:
I would check to make sure that this is an integer. Af far as i can tell your data model doesn't have that check, and just assumes an instance of `int`

Reply:
the method and value are checked on creation and update [here](https://github.com/quay/quay/pull/2393/files#diff-0445600d80e07f49e4ac5d97b6853caebdeec7024679044c9c62c3b8eb31154bR132)

Reply:
Actually, never mind, I see that it's essentially a union of either str or int

Reply:
Do you see having other possible types in the future? If that's the case, it might be worthwhile to wrap this in it's own datatype.

Reply:
How so?

Reply:
Similar to how we define datatypes in `data/registry_model/datatypes.py`. Say if at some point, we end up with some kind of dict as the value type.

But this should be fine as is for now.

Reply:
yeah you're right, this will have to be apart of the future policy implementations.